src/main/java/graphql/execution/ExecutionStepInfo.java

```diff
@@ -299,11 +299,8 @@ public Builder path(ResultPath resultPath) {
             return this;
         }
 
-        public Builder arguments(Supplier<Map<String, Object>> arguments) {
-            this.arguments = () -> {
-                Map<String, Object> map = arguments.get();
-                return map == null ? ImmutableMapWithNullValues.emptyMap() : ImmutableMapWithNullValues.copyOf(map);
-            };
+        public Builder arguments(Supplier<ImmutableMapWithNullValues<String, Object>> arguments) {
```

Comment:
Technically this is a breaking change

I was going to leave the old and deprecate it and make a new one - but why - we dont really expect any one to build ESI objects and hence they can fix it up

the only caller of this is one place in the ES code

If you object - I can leave the old - I actually had that and then say "nah yeah nah"